src/examples/algorithms/VectorHelpers.hpp

```diff
@@ -0,0 +1,105 @@
+#ifndef NTA_UTILS_VECTORHELPERS
+#define NTA_UTILS_VECTORHELPERS
+
+#include <vector>
+#include <string>
+#include <iostream>
+#include <sstream>
+#include <typeinfo>
+
+#include <algorithm>
+
+#include <nupic/types/Types.hpp>
+
+
+namespace nupic {
+namespace utils {
+
+class VectorHelpers
+{
+    public:
+        /**
+        * pretty print a vector v, with separator string sep, to output stream os (default cout)
+        */
+        template<typename T>
+        static void print_vector(const std::vector<T>& v, std::string sep="", std::string prefix="", std::ostream& os=std::cout)
+{
+  os << prefix;
+  for (auto it=v.cbegin(); it != v.cend(); ++it) {
+    os << *it;
+    if ( ((it + 1) != v.cend()) &&  (sep != "") ) {
+      os << sep;
+    }
+  }
+  os << std::endl;
+}
+        /**
+        * cast a vector to a different (compatible) type
+        */
+        template<typename T1, typename T2>
+        static std::vector<T2> castVectorType(const std::vector<T1>& orig)
```

Comment:
if other problems are fixed, this would be only needed for TP, I've suggested moving it there as private method and using this func to make overloaded versions